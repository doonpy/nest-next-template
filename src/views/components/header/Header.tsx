import Head from 'next/head';
import React from 'react';
import { useSelector } from 'react-redux';

import PageReducer from '../../redux/reducers/PageReducer';

const Header: React.FC = () => {
  const pageReducer = PageReducer.getInstance();
  const pageState = useSelector<RootState, PageState>(
    (state) => state[pageReducer.getReducerKey()]
  );

  return (
    <Head>
      <title>{pageState.title}</title>
    </Head>
  );
};

export default Header;
