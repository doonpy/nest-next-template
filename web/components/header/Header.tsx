import Head from 'next/head';
import React from 'react';

interface ComponentProps {
  title: string;
}

const Header: React.FC<ComponentProps> = ({ title }) => {
  return (
    <Head>
      <title>{title}</title>
    </Head>
  );
};

export default Header;
