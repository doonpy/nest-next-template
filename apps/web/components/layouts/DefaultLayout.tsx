import React from 'react';

import Header from '../header/Header';

const DefaultLayout: React.FC<PageState> = ({ children, title }) => {
  return (
    <>
      <Header title={title} />
      {children}
    </>
  );
};

export default DefaultLayout;
