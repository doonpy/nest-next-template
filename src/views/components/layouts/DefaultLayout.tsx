import React from 'react';

import Header from '../header/Header';

const DefaultLayout: React.FC = ({ children }) => {
  return (
    <>
      <Header />
      {children}
    </>
  );
};

export default DefaultLayout;
