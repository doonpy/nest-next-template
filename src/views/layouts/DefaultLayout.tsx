import React from 'react';

import Header from '../components/header/Header';

interface LayoutProps extends CommonPageProps {
  [key: string]: any;
}

const DefaultLayout: React.FC<LayoutProps> = ({ children, ...props }) => {
  return (
    <>
      <Header />
      {children}
    </>
  );
};

export default DefaultLayout;
