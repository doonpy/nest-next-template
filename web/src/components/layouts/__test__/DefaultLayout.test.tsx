import { shallow } from 'enzyme';
import React from 'react';

import Header from '../../header/Header';
import DefaultLayout from '../DefaultLayout';

describe('DefaultLayout', () => {
  const title = 'test title';
  const children = <div>Children</div>;

  it('should render correctly', () => {
    const component = shallow(<DefaultLayout title={title}>{children}</DefaultLayout>);

    expect(component).toMatchSnapshot();
  });

  it('should contains correctly children', () => {
    const headerChildren = <Header title={title} />;
    const component = shallow(<DefaultLayout title={title}>{children}</DefaultLayout>);

    expect(component.contains(children)).toBeTruthy();
    expect(component.contains(headerChildren)).toBeTruthy();
  });
});
