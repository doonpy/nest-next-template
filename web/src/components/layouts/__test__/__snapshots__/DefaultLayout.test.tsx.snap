// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout should render correctly 1`] = `
<Fragment>
  <Header
    title="test title"
  />
  <div>
    Children
  </div>
</Fragment>
`;
